### PR TITLE
[civ2][docker/1-1] rename ray_ci binaries to test_in_docker

### PR DESCRIPTION
**Pull request by can-anyscale**
Rename a couple of files:
ray_ci -> test_in_docker
runner -> tester
test_runner -> test_tester

Test:
- CI

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]
lgtm!

**Comment by can-anyscale** [comment]
@scottjlee can i have your stamp on this PR again - i merged https://github.com/ray-project/ray/pull/39042 into this PR so I can land the whole stack and now it requires approval from code owner again, thanks

**Review by scottjlee — APPROVED** [review]

**Pull request merged**
